index.js

```diff
@@ -58,13 +58,45 @@ module.exports = (src, dest, opts) => {
 	return promise;
 };
 
-module.exports.sync = function (src, dest, opts) {
-	if (!src || !dest) {
-		throw new CpFileError('`src` and `dest` required');
+const checkSrcIsFile = function (src) {
+	const stat = fs.statSync(src);
+	if (stat.isDirectory()) {
+		throw Object.assign(new CpFileError(`EISDIR: illegal operation on a directory '${src}'`), {
+			errno: -21,
+			code: 'EISDIR',
+			src
+		});
 	}
+};
 
-	opts = Object.assign({overwrite: true}, opts);
+const fixupTimestampAndClose = function (read, write, src, dest) {
+	const stat = fs.fstatSync(read, src);
+	fs.futimesSync(write, stat.atime, stat.mtime, dest);
+	fs.closeSync(read);
+	fs.closeSync(write);
+};
+
+const copyNative = function (src, dest, opts) {
+	checkSrcIsFile(src);
+	fs.makeDirSync(path.dirname(dest));
+
+	const flags = opts.overwrite ? 0 : fs.constants.COPYFILE_EXCL;
+	try {
+		fs.copyFileSync(src, dest, flags);
+	} catch (err) {
+		if (!opts.overwrite && err.code === 'EEXIST') {
+			return;
+		}
 
+		throw err;
+	}
+
+	const read = fs.openSync(src, 'r');
+	const write = fs.openSync(dest, 'r+');
```

Comment:
I don't think we need these at all since we aren't using the file descriptors for other than setting `utimes`. We can do that anyway using [`fs.utimesSync`](https://nodejs.org/api/fs.html#fs_fs_utimessync_path_atime_mtime).